homeassistant/components/samsungtv/media_player.py

```diff
@@ -325,7 +325,8 @@ def _on_upnp_event(
         self, service: UpnpService, state_variables: Sequence[UpnpStateVariable]
     ) -> None:
         """State variable(s) changed, let home-assistant know."""
-        if self._update_from_upnp():
+        # Ensure the entity has been added to hass to avoid race condition
+        if self._update_from_upnp() and self.entity_id:
```

Comment:
Entity callbacks should normally be registered in `async_added_to_hass`. That avoids problems like this.